backend/compact-connect/stacks/transaction_monitoring_stack/transaction_history_processing_workflow.py

```diff
@@ -0,0 +1,236 @@
+        # grant access to the compact specific secrets manager secrets following this namespace pattern
+        # compact-connect/env/{environment_name}/compact/{compact_name}/credentials/payment-processor
+        self.transaction_processor_handler.add_to_role_policy(
+            PolicyStatement(
+                effect=Effect.ALLOW,
+                actions=[
+                    'secretsmanager:GetSecretValue',
+                ],
+                resources=[
+                    self._get_secrets_manager_compact_payment_processor_arn_for_compact(
+                        compact=compact, environment_name=environment_name
+                    )
+                ],
+            )
+        )
+        NagSuppressions.add_resource_suppressions_by_path(
+            stack=stack,
+            path=f'{self.transaction_processor_handler.node.path}/ServiceRole/DefaultPolicy/Resource',
+            suppressions=[
+                {
+                    'id': 'AwsSolutions-IAM5',
+                    'reason': """
+                            This policy contains wild-carded actions and resources but they are scoped to the
+                            specific actions, KMS key, and Table that this lambda needs access to.
+                            """,
+                },
+            ],
+        )
+
+        # Create Step Function definition
+        # set initial values for the process task to use
+        self.initialize_state = Pass(
+            self,
+            f'{compact}-InitializeState',
+            result=Result.from_object({'Payload': {'compact': compact, 'processedBatchIds': []}}),
+        )
+
+        self.processor_task = LambdaInvoke(
+            self,
+            f'{compact}-ProcessTransactionHistory',
+            lambda_function=self.transaction_processor_handler,
+            input_path=JsonPath.string_at('$.Payload'),
+            result_path='$',
+            task_timeout=Timeout.duration(Duration.minutes(15)),
+        )
+        self.check_status = Choice(self, f'{compact}-CheckProcessingStatus')
+
+        self.email_notification_service_invoke_step = LambdaInvoke(
+            self,
+            f'{compact}-BatchFailureNotification',
+            lambda_function=persistent_stack.email_notification_service_lambda,
+            payload=TaskInput.from_object(
+                {
+                    'compact': compact,
+                    'template': 'transactionBatchSettlementFailure',
+                    'recipientType': 'COMPACT_OPERATIONS_TEAM',
+                }
+            ),
+            result_path='$.notificationResult',
+            task_timeout=Timeout.duration(Duration.minutes(15)),
+        )
+
+        success = Succeed(self, f'{compact}-ProcessingComplete')
+        fail = Fail(self, f'{compact}-ProcessingFailed', cause='Transaction processing failed')
+
+        # Here we define the chaining between the steps in the state machine
+        self.initialize_state.next(self.processor_task)
```

Comment:
Thanks for adding the diagram for this. Can you adjust the diagram to make the correlation to this code a little more obvious? I think just adjusting the two decision nodes would suffice. This flow looks more like one decision node based on status message, `COMPLETE`, `IN_PROGRESS`, `BATCH_FAILURE`, and I guess <anything else> for your `Fail` case. Your choice here how to make it more obvious if you prefer to have represent this singe Choice step as two decisions with a logical flow that explains a bit how the states coming out of TransactionHistoryProcessor work, but it just needs to correlate to this code a bit more, to be evident to somebody new.

Reply:
Good call. I've updated the diagram to reflect the single choice step.